### PR TITLE
Fix #2809, modify attention

### LINKED ISSUE #2809: Mismatch in RoPE and RMSNorm Ordering Between TorchTune and HuggingFace Qwen3 Implementations
Hi!

I noticed that the logits produced by TorchTune's Qwen3 implementation differ from those generated by the HuggingFace Qwen3 model.

It appears the discrepancy stems from the relative ordering of RoPE and RMSNorm in the attention module:

- In torchtune, [{q,k}_norm is applied after RoPE](https://github.com/pytorch/torchtune/blob/23b3f7b421ff891c782d021021fed328c6509adc/torchtune/modules/attention.py#L242).
- In HuggingFace's transformer, [{q,k}_norm is applied before RoPE](https://github.com/huggingface/transformers/blob/81799d8b556b3c810ed314187674bc439c0582b4/src/transformers/models/qwen3/modeling_qwen3.py#L206).

This might be a broader issue, as torchtune uses a unified TransformerDecoder implementation for many Transformer-based models, which potentially differ in the exact details of their forward passes.

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [X] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* #2809

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
сс @pbontrager

**Comment by pbontrager** [comment]
Sorry this slipped through. This all looks good. Can you verify that you get reasonable results when you prompt the model?
